app/src/main/kotlin/com/wire/android/ui/calling/OngoingCallViewModel.kt

```diff
@@ -40,8 +49,11 @@ class OngoingCallViewModel @Inject constructor(
         .get<String>(EXTRA_CONVERSATION_ID)!!
         .parseIntoQualifiedID()
 
+    private val viewModelJob = SupervisorJob()
+    private val uiScope = CoroutineScope(Dispatchers.Main + viewModelJob)
```

Comment:
what is the reason we are creating custom scope ? the jobs are not going to be cancelled when the ViewModel gets discarded here, also I think SuperVisorJob() does not do much here, since we are not using any custom logic to catch the errrors, it will get propagated upto Java exception handler and just crash the app like we were not using SuperVisorJob()

Reply:
When the user puts the app in background, we need to pause the video..
Since `viewModelScope` is being cleared at the moment at the app is no longer on foreground, I created this custom scope and in onClear I paused the video and canceled the job

Reply:
I am not 100 % certain but from what I know view model is not onCleared when the app in the background it will be if system decide to kill the app process

Reply:
yeah the viewmodel will follow the  Activity/Fragment lifecycle, it goes away on Activity destroyed or Fragment detached.
But with Compose, AFAIK the viewModel scope will be removed as soon as it's removed from the navigation stack.

I've already debugged this, onCleared is immediately called once I put the app in the background

Reply:
that is really weird in my opinion this should not happen I checked just to make sure in several viewModels and onCleared is only called on killing the app and navigating back from the composable

Reply:
It was my bad...I was enabling "don't keep activities" from developer options